usmqe/api/tendrlapi/common.py

```diff
@@ -1,19 +1,96 @@
-"""
-Tendrl REST API common functions.
-"""
+"""Tendrl REST API common functions."""
 
+import json
 import time
+
 import pytest
 import requests
+
 from usmqe.api.base import ApiBase
 
 LOGGER = pytest.get_logger("commonapi", module=True)
 
 
+class TendrlAuth(requests.auth.AuthBase):
+    """
+    Implementation of Tendrl Auth Method (Bearer Token) for requests
+    library, based on upstream documentation:
+
+    https://github.com/Tendrl/api/blob/master/docs/authentication.adoc
+
+    This way, auth. implementation is stored in a single place and it also
+    makes possible to omit auth object entrirely (thanks to design of
+    requests library) for testing purposes (eg. checks of corner cases to make
+    sure that security is not compromised) no matter if it makes sense from
+    Tendrl user perspective.
+
+    See also: http://docs.python-requests.org/en/master/user/authentication/
+    """
+
+    def __init__(self, token, username=None):
+        self._bearer_token = token
```

Comment:
I think in this case it will be better to use two underscores in the variable name, one has slightly different meaning and doesn't protect much. Well, more precise it doesn't protect at all, it just says it should not be changed.